x-pack/plugin/deprecation/qa/rest/src/javaRestTest/java/org/elasticsearch/xpack/deprecation/DeprecationHttpIT.java

```diff
@@ -629,6 +629,81 @@ public void testCompatibleMessagesCanBeIndexed() throws Exception {
         }
     }
 
+    /**
+     * Check that deprecation messages can be recorded to an index
+     */
+    public void testDeprecationIndexingCacheReset() throws Exception {
+        try {
+            configureWriteDeprecationLogsToIndex(true);
+
+            final Request getRequest = createTestRequest("GET");
+            assertOK(client().performRequest(getRequest));
+
+            client().performRequest(new Request("DELETE", "/_logging/deprecation_cache"));
+
+            assertOK(client().performRequest(getRequest));
+
+            assertBusy(() -> {
+                Response response;
+                try {
+                    client().performRequest(new Request("POST", "/" + DATA_STREAM_NAME + "/_refresh?ignore_unavailable=true"));
+                    response = client().performRequest(new Request("GET", "/" + DATA_STREAM_NAME + "/_search"));
+                } catch (Exception e) {
+                    // It can take a moment for the index to be created. If it doesn't exist then the client
+                    // throws an exception. Translate it into an assertion error so that assertBusy() will
+                    // continue trying.
+                    throw new AssertionError(e);
+                }
+                assertOK(response);
+
+                ObjectMapper mapper = new ObjectMapper();
+                final JsonNode jsonNode = mapper.readTree(response.getEntity().getContent());
+
+                final int hits = jsonNode.at("/hits/total/value").intValue();
+                assertThat(hits, greaterThan(0));
+
+                List<Map<String, Object>> documents = new ArrayList<>();
+
+                for (int i = 0; i < hits; i++) {
+                    final JsonNode hit = jsonNode.at("/hits/hits/" + i + "/_source");
+
+                    final Map<String, Object> document = new HashMap<>();
+                    hit.fields().forEachRemaining(entry -> document.put(entry.getKey(), entry.getValue().textValue()));
+
+                    documents.add(document);
+                }
+
+                logger.warn(documents);
+                assertThat(documents, hasSize(4));
+
+                assertThat(
+                    documents,
+                    hasItems(
```

Comment:
I don't understand why we have 3 items of one shape, and 1 with a different one. I could understand if we had 2 of each. Can you explain for me, since I must be missing something?

Reply:
great find.. This is a copy paste bug and it was passing only because hasitmes was used instead of contains. 
I repalced `hasItems` in all testcases here with `containsInAnyOrder` to prevent any further mistakes